include/kernels/dslash_misc.cuh

```diff
@@ -0,0 +1,314 @@
+#pragma once
+
```

Comment:
Since this seems to be a bit of a grab bag I wonder whether it might be better to split that in individual files with specific names, e.g. dslash_<name>_helpers with name = {shmem, twist, clover} ? Might make code navigation easier.

Reply:
Good idea.  206e7113ed71a2daec501161c41acbec314b4fcc does this.